migrations/20190603225918-no-paranoid.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * migrations/20190603225918-no-paranoid.js
+ *
+ * Note: the *new* indices will get created on db startup *without* any
+ * migration code, which means they cannot be created as *unique* until
+ * *after* the "destroySoftDeleted" step has completed!
+ */
+'use strict';
+const Promise = require('bluebird');
+const db = require('../db/index');
+const IS = 'isDeleted';
+const AT = 'deletedAt';
+
+/**
+ * If a collector references a soft-deleted collector group, clear that
+ * collector group from the collector.
+ *
+ * If a perspective references a soft-deleted lens, delete the perspective
+ * (since it obviously has not been used).
+ *
+ * @param qi
+ * @param Seq
+ * @returns {*}
+ */
+function fixReferences(qi) {
+  console.log('fixReferences...');
+  return qi.sequelize.query(
+    `UPDATE "Collectors" SET "collectorGroupId" = NULL ` +
+    `WHERE "collectorGroupId" IN ` +
+    `  (SELECT id FROM "CollectorGroups" WHERE "isDeleted" > 0)`
+  )
+    .catch((err) => console.log(` [ERR] fixReferences: ${err.message}`))
+    .then((n) => console.log(` [OK] fixReferences cleared soft-deleted ` +
+      `collectorGroupId from Collectors records: ${JSON.stringify(n)}`))
+    .then(() => qi.sequelize.query(`DELETE FROM "Perspectives" ` +
+      `WHERE "lensId" IN (SELECT id FROM "Lenses" WHERE "isDeleted" > 0)`))
+    .catch((err) => console.log(` [ERR] fixReferences: ${err.message}`))
+    .then((n) => console.log(` [OK] fixReferences deleted Perspectives ` +
+      `which have soft-deleted lensId: ${JSON.stringify(n)}`))
+    .then(() => console.log('fixReferences... done!\n'));
+} // fixReferences
+
+function destroySoftDeleted(qi, Seq) {
+  const optsIs = {
+    where: { isDeleted: { [Seq.Op.gt]: 0 } },
+    force: true,
+  };
+  const optsAt = {
+    where: { deletedAt: { [Seq.Op.ne]: null } },
+    force: true,
+  };
+  const exec = (modelName, opts) => db[modelName].destroy(opts)
+    .then((n) => console.log(` [OK] destroySoftDeleted ${modelName}: ${n}`))
+    .catch((err) =>
+      console.log(` [ERR] destroySoftDeleted ${modelName}: ${err.message}`));
+
+  console.log('destroySoftDeleted...');
+
+  /*
+   * Why this order?
+   * - Generator THEN Collector THEN CollectorGroup so we don't try to delete
+   *   a CollectorGroup which is referenced by a Collector or Generator
+   * - Generator THEN GeneratorTemplate so we don't try to delete a
+   *   GeneratorTemplate which is referenced by a Generator
+   * - Perpsective THEN Lens/Subject so we don't try to delete a Lens or
+   *   Subject which is referenced by a Perspective
+   */
```

Comment:
>> Promise.all will run the promises in parallel, so it's possible they won't be executed in the right order.

> Yeah at one point I had a sequence of Promise.alls but then I wrapped the whole list in a trx so it all happens atomically.

Does that mean the order doesn't matter here anymore? If so can you update the comment?